RocketChat/Rocket.Chat
# fix: Incorrect Email Validation and Persistent display of error message on message selection in MailExportForm #32974

## Proposed changes (including videos or screenshots)

With respect to the `Export Message` dialog box 
1. Have introduced changes in validateEmail function in order to validate email according to RFC standards.
2. Remove persistent error message if no message is selected. 

## Issue(s)

Closes #32974

## Steps to test or reproduce

1. Enter an invalid email like 'zack@gmail' which would have allowed you to export the message earlier although the email was invalid. Now you have to enter a valid email. 
2. If no message was selected then an error tray is shown but it persisted even after a user selected a message.With this change, the error would not be persisted. 

## Further comments
